ir/pattern/constraint.rs

```diff
@@ -510,6 +511,7 @@ pub enum Constraint<ID> {
     Plays(Plays<ID>),
     Value(Value<ID>),
     LinksDeduplication(LinksDeduplication<ID>),
+    OptimisedToUnsatisfiable(OptimisedToUnsatisfiable),
```

Comment:
`Constraint::Unsatisfiable` Optimisation irrelevant to this data structure. The constraint can also just be `Unsatisfiable`